### PR TITLE
feat: add maven-workspace plugin

### PR DESCRIPTION
**Pull request by chingor13**
This plugin analyzes a Maven monorepo's releases and version bumps any interdependencies. It behaves similarly to the `node-workspace` and `cargo-workspace` plugins.

Example PR: https://github.com/googleapis/java-cloud-bom/pull/4335

feat: PomXml updater can update dependency versions

### CODE REVIEW

**Comment by chingor13** [comment]
cc @suztomo

File: src/plugins/maven-workspace.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+      logger.warn(`Missing project.version in ${path}`);
+      return;
+    }
+    const dependencies: Gav[] = [];
+    const testDependencies: Gav[] = [];
+    for (const dependencyNode of xpath.select(
+      XPATH_PROJECT_DEPENDENCIES,
+      document
+    ) as Node[]) {
+      const parsedNode = parseDependencyNode(dependencyNode);
```
Comment (bcoe) [inline src/plugins/maven-workspace.ts:96]: Is the assumption that this list of dependencies represents local dependencies within a monorepo?

**Review by bcoe — APPROVED** [review]
I understanding the algorithm correctly that it:

1. loads a list of the packages handled in the monorepo.
1. searches for all POM files in the repo.
1. for each POM file, looks at the `dependencies` stanza, if this artifact name lines up with libraries found in `1.`, updates the dependency?
3.

File: src/plugins/maven-workspace.ts (hunk 1 above)
Comment (chingor13) [reply src/plugins/maven-workspace.ts:96]: This is all dependencies, but is only relevant when building the local dependency graph.

**Comment by suztomo** [comment]
Thank you!

**Pull request merged**